src/python/pants/backend/python/dependency_inference/module_mapper.py

```diff
@@ -308,6 +311,24 @@ def providers_for_module(
         )
 
 
+@functools.cache
+def generate_mappings_from_pattern(proj_name: str, fallback: str) -> Iterable[str]:
+    """Generate a list of possible module mappings from a project name using a regex pattern.
+
+    e.g. google-cloud-foo -> [google.cloud.foo, google.cloud.foo_v1, google.cloud.foo_v1beta1]
+    Should eliminate the need to "manually" add a mapping for every service
+    proj_name: The project name to generate mappings for e.g google-cloud-datastream
+    """
+    pattern_values = []
+    for match_pattern, replace_patterns in DEFAULT_MODULE_PATTERN_MAPPING.items():
+        if match_pattern.match(proj_name) is not None:
```

Comment:
**question:** stop after the first match.
right now if a particular project matches multiple patterns, then all of them will be applied.
curious what folks thoughts are on this?
an example:
```python
{
  re.compile("bob"): [do_something],
  re.compile(".*bob.*"): [do_another_something],
}
```
in this case we'd get replacement patterns that are generate by both `do_something` and `do_another_something`. curious what folks thoughts are on this behavior.

if the patterns are identical like:
```python
{
  re.compile("bob"): [do_something],
  re.compile("bob"): [do_another_something],
}
```
then only one will be done and i believe it's the last one. this is because `re.compile("bob")` always generates to the same key so the dict will only have one in there.

Reply:
Hm, good question. In some sense, it seems like it should be an error in the module mappings if patterns overlapped, so I'd be fine to just stop after the first match. 

I don't know if it's worth the effort to actually validate non-overlapping-ness in a pants test. I imagine one could theoretically intersect the regexes and then see if any match exists, e.g. with https://pypi.org/project/interegular/ .